### PR TITLE
Results page

### PR DESCRIPTION
**Pull request by tpkelly**
Plots the results of a poll using InsightJS
Fixes the links going between the vote and results pages
Uses the proper expiry date + poll question/name in the poll layout

### CODE REVIEW

File: VotingApplication/VotingApplication.Web/Scripts/Controllers/HomePageController.js [hunk 1]
```diff
@@ -1,10 +1,65 @@
...
-    VotingApp.controller('HomePageController', function ($scope) {
+    VotingApp.controller('HomePageController', ['$scope', 'PollService', function ($scope, PollService) {
         $scope.models = {
             pageTitle: 'Pollster'
         };
-    });
+
+        $scope.pollExpiry = undefined;
+
+        var timeStringForTimeout = function (expiryDate, timeUnit, timeUnitName) {
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web/Scripts/Controllers/HomePageController.js]: Both this and the calculateExpiry functions could probably be moved out of this controller. I anticipate that the poll management page might want to also show this information at a later date.

File: VotingApplication/VotingApplication.Web/Scripts/Controllers/ResultsPageController.js [hunk 2]
```diff
@@ -1,5 +1,123 @@
...
+                var chartSeries = chart ? chart.series() : null;
+                dataUnchanged = dataUnchanged &&
+                    chart && chart.series().length > n &&
+                    JSON.stringify(data[n].Data) == JSON.stringify(chart.series()[n].data.rawData());
+            }
+            //Exit early if data has not changed
+            if (dataUnchanged)
+                return;
+
+            // Hack to fix insight's lack of data reloading
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web/Scripts/Controllers/ResultsPageController.js]: Can remove this

File: VotingApplication/VotingApplication.Web/Views/Home/Index.cshtml [hunk 3]
```diff
@@ -7,8 +7,8 @@
 <body>
     <div class="center-panel">
         <div class="body-content">
-            <h2>Where do we go for lunch?</h2>
-            <p id="expiry-date">Poll closes in: 5 hours</p>
+            <h2>{{pollName}}</h2>
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web/Views/Home/Index.cshtml]: Small point, bit if I recall correctly, we will encourage the poll name to be a question by putting a question mark after the name of the poll during poll creation, so perhaps one should be put here?
i.e.
{{pollName}}?

Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Views/Home/Index.cshtml]: The question mark is going to be added at the poll creation stage (mostly for backwards compatibility here when calling up old polls).

**Pull request merged**